hardware_interface/test/fake_components/test_generic_system.cpp

```diff
@@ -1027,3 +1100,176 @@ TEST_F(TestGenericSystem, generic_system_2dof_functionality_with_offset_custom_i
   EXPECT_EQ(
     states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
 }
+
+TEST_F(TestGenericSystem, valid_urdf_ros2_control_system_robot_with_gpio_)
+{
+  auto urdf = ros2_control_test_assets::urdf_head +
+              valid_urdf_ros2_control_system_robot_with_gpio_ + ros2_control_test_assets::urdf_tail;
+  hardware_interface::ResourceManager rm(urdf);
+
+  // check is hardware is configured
+  std::unordered_map<std::string, rclcpp_lifecycle::State> states_map;
+  states_map = rm.get_components_states();
+  EXPECT_EQ(
+    states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
+
```

Comment:
I am not really trusting our code about exporting correct command and state interfaces properly. Can you here, please add check of number of command and state interfaces? When "fake_gpio_command_interfaces_" is `false` then we should have something like this:

```suggestion

ASSERT_EQ(rm.state_interface_keys().size(), 10);
ASSERT_EQ(rm.command_interface_keys().size(), 6);

```

For another case, see the comment below.